cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -229,15 +288,12 @@ protected void rebalanceForBroker(Broker broker,
                                     ClusterModel clusterModel,
                                     Set<Goal> optimizedGoals,
                                     Set<String> excludedTopics)
-      throws AnalysisInputException, ModelInputException {
+      throws ModelInputException {
     boolean requireLessLoad = !isLoadUnderBalanceUpperLimit(clusterModel, broker);
     boolean requireMoreLoad = !isLoadAboveBalanceLowerLimit(clusterModel, broker);
     if (broker.isAlive() && !requireMoreLoad && !requireLessLoad) {
       // return if the broker is already under limit.
       return;
-    } else if (!clusterModel.newBrokers().isEmpty() && requireMoreLoad && !broker.isNew()) {
```

Comment:
Why are we removing this?

Reply:
Because while it is not possible to use `rebalanceByMovingLoadIn` to get more load to an old broker in case there are new brokers, it is potentially possible to swap load using `rebalanceBySwappingLoadIn` between an old broker and a new broker if the replica, which currently resides in the new broker, was originally located in the old broker.
Note that 
- `rebalanceByMovingLoadIn` (via `maybeApplyBalancingAction` > `getEligibleBrokers`) will already disallow moving replicas to an old broker if there are new brokers. After this feedback, to make this cutoff earlier, I copied the logic from within `getEligibleBrokers` to `rebalanceByMovingLoadIn`.
- `rebalanceBySwappingLoadIn` will allow swapping only if the conditions in the paragraph above are satisfied.